src/OpenIddict.Core/OpenIddictController.cs

```diff
@@ -147,6 +182,11 @@ public class OpenIddictController<TUser, TApplication> : Controller where TUser
             identity.Actor.AddClaim(ClaimTypes.NameIdentifier, request.ClientId);
             identity.Actor.AddClaim(ClaimTypes.Name, await Manager.GetDisplayNameAsync(application), destination: "id_token token");
 
+            // Add selected scopes back to ASOS
+            if (Request.Form.ContainsKey("scope")) {
+                request.Scope = Request.Form["scope"];
```

Comment:
FYI, I just pushed a commit that should help you here: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/samples/Mvc/Mvc.Server/Controllers/AuthorizationController.cs#L132-L144